lib/bank_account/account.rb

```diff
@@ -0,0 +1,44 @@
+module BankAccount
+  class Account
+  attr_accessor :id, :initial_balance
+
+    def initialize(id, initial_balance)
+      ## optimize raise_error, initialize
+      if initial_balance < 0
+        raise ArgumentError.new("You cannot open an account with a negative balance.")
+      else
+      @id = id.round(2)
+      @initial_balance = initial_balance.round(2)
+      @balance = initial_balance.round(2)
+      end
+    end
+
+
+    def balance
+      return @balance
+    end
+
+    def withdraw(amount)
+      amount = amount.round(2)
+      if amount > @balance
+        puts "Nonsufficient funds."
+        balance
+      else
+      @balance -= amount
+      balance
+      end
+    end
+
+    def deposit(amount)
+      amount = amount.round(2)
```

Comment:
Since you're using `.round(2)` in several places you might want to create a method that handles this rounding instead. Then, if you needed to change the number of decimal places your system supported you would only need to change it in one place rather than multiple.